src/heitech.zer0mqXt.core/Main/Bus.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using heitech.zer0mqXt.core.infrastructure;
+using heitech.zer0mqXt.core.patterns;
+
+namespace heitech.zer0mqXt.core.Main
+{
+    internal class Bus : IEntry
+    {
+        private readonly RqRep _rqRep;
+        private readonly PubSub _pubSub;
+        public Bus(SocketConfiguration config) 
+        {
+            _rqRep = new RqRep(config);
+            _pubSub = new PubSub(config);
+        }
+
+        public void Dispose()
+        {
+            _rqRep.Dispose();
+            _pubSub.Dispose();
+        }
+
+        public async Task PublishAsync<TMessage>(TMessage message) 
+            where TMessage : class, new()
+        {
+            var result = await _pubSub.PublishAsync(message);
```

Comment:
One thought that hits me right now: should we use `ConfigureAwait(false)` everywhere?
Because this is a reusable library which could be consumed by an app model which has a custom `SynchronizationContext` and/or a non-default `TaskScheduler`.

If you agree - I wouldn't mind creating an issue for that. Could be my first code contribution 😉

Reply:
Yes this is a great idea !